app/models/concerns/hyrax/collection_behavior.rb

```diff
@@ -85,6 +86,16 @@ def to_s
       title.present? ? title.join(' | ') : 'No Title'
     end
 
+    def total_file_sets
+      files = []
+      member_works.map { |work| files.push(work.file_sets.count) }
+      files.sum
+    end
+
+    def pinned?(user_id)
+      UserPinnedCollection.find_by(user_id: user_id, collection_id: id)
```

Comment:
if we keep this here use `exists?` instead of `find_by` to avoid pulling data from the DB unnecessarily.

```suggestion
      UserPinnedCollection.exists?(user_id: user_id, collection_id: id)
```